sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -157,6 +157,8 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession {
   protected override def sparkConf: SparkConf = super.sparkConf
     // Fewer shuffle partitions to speed up testing.
     .set(SQLConf.SHUFFLE_PARTITIONS, 4)
+    // Disable BHJ optimization to avoid changing result order when we enable AQE
+    .set(SQLConf.AUTO_BROADCASTJOIN_THRESHOLD, -1.toLong)
```

Comment:
Why don't we just disable AQE explicitly here? Do we have SQL-file based test cases for AQE?

Reply:
Because we still want to test it with AQE as it's good for AQE to improve test coverage.

We don't have separate SQL-file based test cases for AQE. AQE is only a way of optimization, so it ideally should not have any specific requirements for the query itself.

Reply:
It's a bit odd because we're working around by another configuration to disable AQE here. But okay since we commented.

Reply:
Maybe we could have `SQLQueryTestSuiteAQE` and override `blackList` to disable some queries? But the problem is `blackList` now can only disable a whole sql file rather than a single query. Maybe we can extend it if we think it's necessary for AQE in the future. @HyukjinKwon @cloud-fan @maryannxue

Reply:
This won't reduce test coverage IMO, as we set the broadcast threshold explicitly when we want to test different physical joins.

Reply:
Okay, lgtm for now.